app/components/userOnboarding/UserDetails.tsx

```diff
@@ -1,19 +1,51 @@
-import React from 'react';
+import React, { useState } from 'react';
 import UserInput from '../common/userInput';
 import Dropdown from '../common/dropdown';
 
-const UserDetails = () => (
-  <div className="mx-4">
-    <div>
-      <UserInput label="Username" placeholder="username here" link="hap.day/" />
-    </div>
-    <div>
-      <UserInput label="Full name" placeholder="Jane doe" />
-    </div>
-    <div>
-      <Dropdown placeholder="select timezone" />
+interface UserDetailsInterface {
+  username: string;
+  fullName: string;
+}
+
+const UserDetails = () => {
+  const initialUserDetails = {
+    username: '',
+    fullName: '',
+  };
+  const [userForm, setUserForm] = useState<UserDetailsInterface>(initialUserDetails);
+
+  const updateUsername = (username: string) => {
+    setUserForm((user) => ({ ...user, username }));
+  };
+
+  const updateFullName = (fullName: string) => {
+    setUserForm((user) => ({ ...user, fullName }));
+  };
+
+  return (
+    <div className="mx-4">
+      <div>
+        <UserInput
+          label="Username"
+          placeholder="username here"
+          link="hap.day/"
+          value={userForm.username}
+          setValue={updateUsername}
+        />
+      </div>
+      <div>
+        <UserInput
+          label="Full name"
+          placeholder="Jane doe"
+          value={userForm.fullName}
+          setValue={updateFullName}
+        />
+      </div>
+      <div>
+        <Dropdown placeholder="select timezone" />
+      </div>
     </div>
-  </div>
-);
+  );
+};
 
 export default UserDetails;
```

Comment:
i think we could have had this in another `PR`, but thanks for this.

Reply:
@harshith-venkatesh

Reply:
> i think we could have had this in another `PR`, but thanks for this.

I updated it as it was also using `UserInput`